src/user/user.service.ts

```diff
@@ -3,25 +3,65 @@ import { UpdateUserDto } from './dto/update-user.dto';
 import { InjectRepository } from '@nestjs/typeorm';
 import { User } from './entities/user.entity';
 import { Repository } from 'typeorm';
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { ResponseUserDTO } from './dto/response-user.dto';
 
 @Injectable()
 export class UserService {
 
   constructor(@InjectRepository(User) private userRepository: Repository<User>) {}
-  async create(createUserDto: CreateUserDto): Promise<User> {
-    const {userName, email, password} = createUserDto;
+  async create(createUserDto: CreateUserDto): Promise<ResponseUserDTO> {
+    const { fullName, userName, dateOfBirth, email, phoneNumber, role, gender , password} = createUserDto;
+
+    const existingUser = await this.userRepository.findOne({
+      where: [{ email }, { user_name: userName }, { phone_number: phoneNumber }],
+    });
+    if (existingUser) {
```

Comment:
Do you have another idea to refactor this. Don't nested the if else statement in many levels.